sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -1154,8 +1154,18 @@ case class CreateExternalRow(children: Seq[Expression], schema: StructType)
 case class EncodeUsingSerializer(child: Expression, kryo: Boolean)
   extends UnaryExpression with NonSQLExpression {
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported")
+  override def eval(input: InternalRow): Any = {
+    val obj = child.eval(input)
+    if (obj == null) return null
+
+    val conf = Option(SparkEnv.get).map(_.conf).getOrElse(new SparkConf)
+    val serializer = if (kryo) {
```

Comment:
Can you add this as a transient lazy field? That will save us from recreating serializer instances all over the place.

Reply:
yes, actually the same could/should be done also for the code generated version, shouldn't it? If so, should I do both changes here or in another PR?

Reply:
If I am not mistaken the code generated version adds the serializer instance to classes immutable state, so there is not need.

Reply:
Yes, you're right, sorry. I just realized after my comment. Sorry.

Reply:
No problem :)